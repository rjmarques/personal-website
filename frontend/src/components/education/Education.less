@import "../../theme.less";

.education {
  .education-entry {
    padding-bottom: @card-spacing;

    &:last-child {
      padding-bottom: 0px;
    }
  }
}
