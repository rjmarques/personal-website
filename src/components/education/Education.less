@import "../../theme.less";

.Education {
  .Education-entry {
    padding-bottom: @card-spacing;

    &:last-child {
      padding-bottom: 0px;
    }
  }
}
