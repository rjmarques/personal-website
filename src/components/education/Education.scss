@use "../../theme.scss" as *;

.education {
  .education-entry {
    padding-bottom: $card-spacing;

    &:last-child {
      padding-bottom: 0px;
    }
  }
}
