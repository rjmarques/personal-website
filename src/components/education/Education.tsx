import React from "react";

import ContentSection from "../common/ContentSection";
import DateLocationEntry from "../common/DateLocationEntry";

import "./Education.scss";

const Education = () => (
  <ContentSection
    title="Education"
    subTitle="My academic background"
    wrapperClass="education"
  >
    <ul className="materialize">
      <li className="education-entry">
        <div className="app-card-wide">
          <h3>
            MSc In Computer Science -{" "}
            <a
              href="https://www.fct.unl.pt/en"
              target="_blank"
              rel="noopener noreferrer"
            >
              New University of Lisbon
            </a>
          </h3>
          <div className="app-card-subtitle">
            <DateLocationEntry
              startingDate="2010"
              endingDate="2012"
              location="Lisbon"
            />
          </div>
          <div>
            Dissertation:{" "}
            <a
              href="https://run.unl.pt/handle/10362/8382"
              target="_blank"
              rel="noopener noreferrer"
            >
              Algorithmic skeleton framework for the orchestration of GPU
              computations
            </a>
          </div>
          <div>
            Source Code:{" "}
            <a
              href="https://bitbucket.org/MarrowTeam/marrow"
              target="_blank"
              rel="noopener noreferrer"
            >
              https://bitbucket.org/MarrowTeam/marrow
            </a>
          </div>
        </div>
      </li>
      <li className="education-entry">
        <div className="app-card-wide">
          <h3>
            BSc In Computer Science -{" "}
            <a
              href="https://www.fct.unl.pt/en"
              target="_blank"
              rel="noopener noreferrer"
            >
              New University of Lisbon
            </a>
          </h3>
          <div className="app-card-subtitle">
            <DateLocationEntry
              startingDate="2007"
              endingDate="2010"
              location="Lisbon"
            />
          </div>
        </div>
      </li>
    </ul>
  </ContentSection>
);

export default Education;
