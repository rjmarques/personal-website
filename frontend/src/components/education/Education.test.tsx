import React from "react";
import renderer from "react-test-renderer";

import Education from "./Education";

describe("renders without crashing", () => {
  it("renders correctly", async () => {
    const component = renderer.create(<Education />);
    expect(component).toMatchSnapshot();
  });
});
