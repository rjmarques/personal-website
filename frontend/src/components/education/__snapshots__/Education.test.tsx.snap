// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders correctly 1`] = `
<section
  className="education"
>
  <div
    className="content"
  >
    <div
      className="content-title"
    >
      <h2>
        Education
      </h2>
      <span
        className="section-sub-title"
      >
        My academic background
      </span>
    </div>
    <ul
      className="materialize"
    >
      <li
        className="education-entry"
      >
        <div
          className="app-card-wide"
        >
          <h3>
            MSc In Computer Science -
             
            <a
              href="https://www.fct.unl.pt/en"
              rel="noopener noreferrer"
              target="_blank"
            >
              New University of Lisbon
            </a>
          </h3>
          <div
            className="app-card-subtitle"
          >
            <div
              className="date-location-entry"
            >
              <span
                className="text"
              >
                <span
                  aria-label="calendar"
                  className="anticon anticon-calendar"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="calendar"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                    />
                  </svg>
                </span>
                 
                2010
                 to 
                2012
              </span>
              <span
                className="text"
              >
                <span
                  aria-label="environment"
                  className="anticon anticon-environment"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="environment"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M854.6 289.1a362.49 362.49 0 00-79.9-115.7 370.83 370.83 0 00-118.2-77.8C610.7 76.6 562.1 67 512 67c-50.1 0-98.7 9.6-144.5 28.5-44.3 18.3-84 44.5-118.2 77.8A363.6 363.6 0 00169.4 289c-19.5 45-29.4 92.8-29.4 142 0 70.6 16.9 140.9 50.1 208.7 26.7 54.5 64 107.6 111 158.1 80.3 86.2 164.5 138.9 188.4 153a43.9 43.9 0 0022.4 6.1c7.8 0 15.5-2 22.4-6.1 23.9-14.1 108.1-66.8 188.4-153 47-50.4 84.3-103.6 111-158.1C867.1 572 884 501.8 884 431.1c0-49.2-9.9-97-29.4-142zM512 880.2c-65.9-41.9-300-207.8-300-449.1 0-77.9 31.1-151.1 87.6-206.3C356.3 169.5 431.7 139 512 139s155.7 30.5 212.4 85.9C780.9 280 812 353.2 812 431.1c0 241.3-234.1 407.2-300 449.1zm0-617.2c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 551c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 439c0-29.9 11.7-58 32.8-79.2C454 338.6 482.1 327 512 327c29.9 0 58 11.6 79.2 32.8C612.4 381 624 409.1 624 439c0 29.9-11.6 58-32.8 79.2z"
                    />
                  </svg>
                </span>
                 
                Lisbon
              </span>
            </div>
          </div>
          <div>
            Dissertation:
             
            <a
              href="https://run.unl.pt/handle/10362/8382"
              rel="noopener noreferrer"
              target="_blank"
            >
              Algorithmic skeleton framework for the orchestration of GPU computations
            </a>
          </div>
          <div>
            Source Code:
             
            <a
              href="https://bitbucket.org/MarrowTeam/marrow"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://bitbucket.org/MarrowTeam/marrow
            </a>
          </div>
        </div>
      </li>
      <li
        className="education-entry"
      >
        <div
          className="app-card-wide"
        >
          <h3>
            BSc In Computer Science -
             
            <a
              href="https://www.fct.unl.pt/en"
              rel="noopener noreferrer"
              target="_blank"
            >
              New University of Lisbon
            </a>
          </h3>
          <div
            className="app-card-subtitle"
          >
            <div
              className="date-location-entry"
            >
              <span
                className="text"
              >
                <span
                  aria-label="calendar"
                  className="anticon anticon-calendar"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="calendar"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                    />
                  </svg>
                </span>
                 
                2007
                 to 
                2010
              </span>
              <span
                className="text"
              >
                <span
                  aria-label="environment"
                  className="anticon anticon-environment"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="environment"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M854.6 289.1a362.49 362.49 0 00-79.9-115.7 370.83 370.83 0 00-118.2-77.8C610.7 76.6 562.1 67 512 67c-50.1 0-98.7 9.6-144.5 28.5-44.3 18.3-84 44.5-118.2 77.8A363.6 363.6 0 00169.4 289c-19.5 45-29.4 92.8-29.4 142 0 70.6 16.9 140.9 50.1 208.7 26.7 54.5 64 107.6 111 158.1 80.3 86.2 164.5 138.9 188.4 153a43.9 43.9 0 0022.4 6.1c7.8 0 15.5-2 22.4-6.1 23.9-14.1 108.1-66.8 188.4-153 47-50.4 84.3-103.6 111-158.1C867.1 572 884 501.8 884 431.1c0-49.2-9.9-97-29.4-142zM512 880.2c-65.9-41.9-300-207.8-300-449.1 0-77.9 31.1-151.1 87.6-206.3C356.3 169.5 431.7 139 512 139s155.7 30.5 212.4 85.9C780.9 280 812 353.2 812 431.1c0 241.3-234.1 407.2-300 449.1zm0-617.2c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 551c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 439c0-29.9 11.7-58 32.8-79.2C454 338.6 482.1 327 512 327c29.9 0 58 11.6 79.2 32.8C612.4 381 624 409.1 624 439c0 29.9-11.6 58-32.8 79.2z"
                    />
                  </svg>
                </span>
                 
                Lisbon
              </span>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</section>
`;
